src/Microsoft.Graph/Models/Extensions/DateTimeZoneExtensions.cs

```diff
@@ -0,0 +1,181 @@
+
+            // When in Local or Utc kind, we can convert to the time zone specified
+            // Which may be a no-op if the value is already in the specified time zone
+            // We don't do this for Unspecified kind, as that would infer converting *from* the local time zone
+            if (dateTime.Kind != DateTimeKind.Unspecified)
+            {
+                dateTime = TimeZoneInfo.ConvertTime(dateTime, timeZoneInfo);
+            }
+
+            DateTimeTimeZone dateTimeTimeZone = new DateTimeTimeZone
+            {
+                DateTime = dateTime.ToString(DateTimeFormat, CultureInfo.InvariantCulture),
+                TimeZone = timeZone
+            };
+
+            return dateTimeTimeZone;
+        }
+
+        /// <summary>
+        /// Converts a DatetimeOffset parameter to its equivalent DateTimeTimeZone Complex Type given its TimeZone
+        /// </summary>
+        /// <param name="dateTimeOffset"></param>
+        /// <param name="timeZone"></param>
+        /// <returns></returns>
+        public static DateTimeTimeZone FromDateTimeOffset(DateTimeOffset dateTimeOffset, string timeZone)
+        {
+            TimeZoneInfo timeZoneInfo = GetTimeZoneInfo(timeZone);
+
+            // Convert the input value to the time zone specified.
+            // If it's already in an offset valid for that time zone, this will be a no-op.
+            dateTimeOffset = TimeZoneInfo.ConvertTime(dateTimeOffset, timeZoneInfo);
+
+            DateTimeTimeZone dateTimeTimeZone = new DateTimeTimeZone
+            {
+                DateTime = dateTimeOffset.ToString(DateTimeFormat, CultureInfo.InvariantCulture),
+                TimeZone = timeZone
+            };
+
+            return dateTimeTimeZone;
+        }
+
+        internal static TimeZoneInfo GetTimeZoneInfo(string timeZone)
+        {
+            // The following is built-in, but is platform dependent.
+            // It will use Windows time zones on Windows, and IANA time zones on other platforms.
+
+            return TimeZoneInfo.FindSystemTimeZoneById(timeZone);
+
+
+            // The following is platform independent, working with either Windows or IANA time zones on any platform.
+            // However, it requires a dependency on https://github.com/mj1856/TimeZoneConverter
+            // See also https://github.com/dotnet/corefx/issues/11897
+            //      and https://github.com/dotnet/corefx/issues/2538
+            //
+            // return TimeZoneConverter.TZConvert.GetTimeZoneInfo(timeZone);
+
+
+            // Note that *neither* of the above options map neatly to the list of time zones given in
+            // https://docs.microsoft.com/graph/api/resources/datetimetimezone
+            // That list seems to include many omissions and some additions and customizations.
+            // Those would also need to be handled here, probably by manually mapping the differences to valid IANA time zones.
```

Comment:
Someone should validate the actual list of time zone IDs one-by-one.  If there are any that aren't valid Windows or IANA identifiers, then a dictionary should be added that maps them back to valid identifiers before giving up.

Reply:
We won't be doing this. Doing timezones correctly should be driven by the workloads underlying our Microsoft Graph clients.